pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -144,6 +147,21 @@ func (c OpenShiftAPIServerOperator) sync() error {
 		return nil
 	}
 
+	// block until extension-apiserver-authentication configmap is available
+	// see https://bugzilla.redhat.com/show_bug.cgi?id=1795163#c19 to check why we have to wait for it
+	// in the future we need to change upstream code to be more dynamic
+	if !c.haveObservedExtensionConfigMap {
+		_, err = c.kubeClient.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get("extension-apiserver-authentication", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			klog.Infof("Waiting for %q configmap in %q namespace to be available", "extension-apiserver-authentication", metav1.NamespaceSystem)
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		c.haveObservedExtensionConfigMap = true
```

Comment:
doing this give us a "past the post" behavior which is desireable for this particular configmap to have a balance with ease and direct client calls.

/approve